--- src/TopRoomBar.cc.orig	2018-07-12 11:33:25.000000000 +0530
+++ src/TopRoomBar.cc	2018-07-12 11:34:20.000000000 +0530
@@ -18,12 +18,12 @@
 #include <QDebug>
 #include <QStyleOption>
 
-#include "Avatar.h"
+#include "ui/Avatar.h"
 #include "Config.h"
-#include "FlatButton.h"
+#include "ui/FlatButton.h"
 #include "MainWindow.h"
-#include "Menu.h"
-#include "OverlayModal.h"
+#include "ui/Menu.h"
+#include "ui/OverlayModal.h"
 #include "TopRoomBar.h"
 #include "Utils.h"
 
