--- src/RoomInfoListItem.cc.orig	2018-07-12 11:25:39.000000000 +0530
+++ src/RoomInfoListItem.cc	2018-07-12 11:26:21.000000000 +0530
@@ -24,11 +24,11 @@
 
 #include "Cache.h"
 #include "Config.h"
-#include "Menu.h"
-#include "Ripple.h"
-#include "RippleOverlay.h"
+#include "ui/Menu.h"
+#include "ui/Ripple.h"
+#include "ui/RippleOverlay.h"
 #include "RoomInfoListItem.h"
-#include "Theme.h"
+#include "ui/Theme.h"
 #include "Utils.h"
 
 constexpr int MaxUnreadCountDisplayed = 99;
