--- interface.h.orig 2018-03-03 08:37:28.000000000 -0800
+++ interface.h	2018-04-14 11:46:47.000000000 -0700
@@ -1587,7 +1587,9 @@
   KERN_TYPE_SHA256CRYPT             = 7400,
   KERN_TYPE_KRB5PA                  = 7500,
   KERN_TYPE_SAPB                    = 7700,
+  KERN_TYPE_SAPB_MANGLED            = 7701,
   KERN_TYPE_SAPG                    = 7800,
+  KERN_TYPE_SAPG_MANGLED            = 7801,
   KERN_TYPE_DRUPAL7                 = 7900,
   KERN_TYPE_SYBASEASE               = 8000,
   KERN_TYPE_NETSCALER               = 8100,
